tf_quant_finance/black_scholes/approximations/american_option.py

```diff
@@ -601,5 +681,63 @@ def _npdf(x):
   return tf.math.exp(-0.5 * x**2) / _SQRT_2_PI
 
 
+def _cbnd(dh, dk, rho):
+  """Compute values for the cumulative standard bivariate normal distribution,
+     specifically for use with Bjerksund-Stensland 2002 which always gives
+     the same value for rho.
+  #### References:
+  [1] Genz, A., Numerical Computation of Rectanglar Bivariate and Trivariate
+        Normal and t Probabilities, 2004
+    http://www.math.wsu.edu/faculty/genz/papers/bvnt.pdf"""
+
+  # Python code adapted from Fortran code and naming convention will align
+  # variables with the variables named in BVND function.
+  # http://www.math.wsu.edu/faculty/genz/software/fort77/tvpack.f
+
+  dtype = rho.dtype
+
+  # Gauss Legendre Points and Weights
+  w = tf.constant([0.01761400713915212, 0.04060142980038694, 0.06267204833410906,
+                   0.08327674157670475, 0.1019301198172404, 0.1181945319615184,
+                   0.1316886384491766, 0.1420961093183821, 0.1491729864726037,
+                   0.1527533871307259], dtype=dtype)
+
+  x = tf.constant([-0.9931285991850949, -0.9639719272779138, -0.9122344282513259,
+                   -0.8391169718222188, -0.7463319064601508, -0.6360536807265150,
+                   -0.5108670019508271, -0.3737060887154196, -0.2277858511416451,
+                   -0.07652652113349733], dtype=dtype)
+
+  h = tf.cast(-dh, dtype=dtype)
+  k = tf.cast(-dk, dtype=dtype)
+  hk = h * k
+
+  # In order to avoid `shape_invariants` argument in the while_loop, set
+  # `bvn` to be a Tensor of zeros of the appropriate shape
+  bvn = tf.zeros_like(hk)
+
+  hs = (h * h + k * k) / 2
+  asr = tf.math.asin(rho)
+
+  def batched_calc(hk, hs, asr, x, w):
+      for _ in range(hk.shape.rank):
+        # Shape x.shape + [1, .., 1]
+        x = tf.expand_dims(x, axis=-1)
+        w = tf.expand_dims(w, axis=-1)
+      # Shape x.shape + asr.shape
+      sn1 = tf.math.sin(asr * (-1 * x + 1) / 2)
```

Comment:
instead of `asr.shape`, let us have `bvn.shape`

Reply:
Do you mean update the comments to say that?  I assumed so and updated accordingly, but let me know if I got that wrong.